pytorch_lightning/plugins/training_type/dp.py

```diff
@@ -66,6 +66,19 @@ def setup(self) -> None:
         self.model_to_device()
         self._model = self._setup_model(LightningParallelModule(self._model))
 
+    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
+        """Moves the batch to the correct device.
+
+        The returned batch is of the same type as the input batch, just
+        having all tensors on the correct device.
+        Args:
+            batch: The batch of samples to move to the correct device
+            device: The target device
+            dataloader_idx: The index of the dataloader to which the batch belongs.
+        """
+        device = device or self.root_device
+        return move_data_to_device(batch, device)
```

Comment:
```suggestion
        return move_data_to_device(batch, device=device or self.root_device)
```

Reply:
Why we don't apply `_apply_batch_transfer_handler` with dp? I believe the `on_before_batch_transfer` could still be applied.

```
        model = self.lightning_module
        device = device or self.root_device
        if model is not None:
            return model._apply_batch_transfer_handler(batch, device=device, dataloader_idx=dataloader_idx)
```